streams/src/test/java/org/apache/kafka/streams/integration/QueryableStateIntegrationTest.java

```diff
@@ -800,6 +811,76 @@ public void shouldBeAbleToQueryMapValuesAfterFilterState() throws Exception {
         }
     }
 
+    @Test
+    public void shouldQueryKeyValueWithCacheBypassed() throws Exception {
```

Comment:
This is the test I used to verify the behavior. Note: this test does not verify that the cache is skipped, just that the query works with the cache skipped. I set breakpoints in the code to additionally verify that the cache is actually skipped.

If we go this route, I'd add unit tests so we can mechanically verify that we're skipping the cache.